release-tools/src/main/java/org/springframework/data/release/jira/Jira.java

```diff
@@ -122,85 +137,259 @@ public Ticket getReleaseTicketFor(ModuleIteration iteration) {
+
+			if (!supports(moduleIteration.getProject())) {
+				continue;
+			}
+
+			createReleaseVersion(moduleIteration, credentials);
+		}
 	}
 
-	private Ticket toTicket(JiraIssue issue) {
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.release.jira.JiraConnector#createReleaseVersion(org.springframework.data.release.model.ModuleIteration, org.springframework.data.release.jira.Credentials)
+	 */
+	@Override
+	public void createReleaseVersion(ModuleIteration moduleIteration, Credentials credentials) {
 
-		JiraIssue.Fields fields = issue.getFields();
+		Assert.notNull(moduleIteration, "ModuleIteration must not be null.");
+		Assert.notNull(credentials, "Credentials must not be null.");
 
-		JiraTicketStatus jiraTicketStatus;
-		if (fields.getStatus() != null && fields.getResolution() != null) {
-			JiraIssue.Status status = fields.getStatus();
-			boolean resolved = status.getStatusCategory().getKey().equals("done");
-			JiraIssue.Resolution resolution = fields.getResolution();
+		Map<String, Object> parameters = newUrlTemplateVariables();
+		HttpHeaders httpHeaders = new HttpHeaders();
+		withAuthorization(credentials, httpHeaders);
 
-			jiraTicketStatus = new JiraTicketStatus(resolved, status.getName(), resolution.getName());
-		} else {
-			jiraTicketStatus = JiraTicketStatus.UNKNOWN;
+		Optional<JiraReleaseVersion> versionsForModuleIteration = findJiraReleaseVersion(moduleIteration);
+
+		if (versionsForModuleIteration.isPresent()) {
+			return;
 		}
+		
+		JiraVersion jiraVersion = new JiraVersion(moduleIteration);
+		logger.log(moduleIteration, "Creating Jira release version %s", jiraVersion);
 
-		return new Ticket(issue.getKey(), fields.getSummary(), jiraTicketStatus);
+		JiraReleaseVersion jiraReleaseVersion = new JiraReleaseVersion();
+		jiraReleaseVersion.setProject(moduleIteration.getProjectKey().getKey());
+		jiraReleaseVersion.setName(jiraVersion.toString());
+		jiraReleaseVersion.setDescription(jiraVersion.getDescription());
+
+		operations.exchange(VERSIONS_TEMPLATE, HttpMethod.POST, new HttpEntity<Object>(jiraReleaseVersion, httpHeaders),
+				JiraReleaseVersion.class, parameters).getBody();
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.release.jira.JiraConnector#findJiraReleaseVersion(org.springframework.data.release.model.ModuleIteration)
+	 */
+	@Cacheable("release-version")
+	public Optional<JiraReleaseVersion> findJiraReleaseVersion(ModuleIteration moduleIteration) {
+
+		JiraVersion jiraVersion = new JiraVersion(moduleIteration);
+		HttpHeaders httpHeaders = new HttpHeaders();
+
+		List<JiraReleaseVersion> versionsForModuleIteration = new ArrayList<>();
+		getReleaseVersions(moduleIteration.toString(), moduleIteration.getProjectKey(), httpHeaders, releaseVersions -> {
+			releaseVersions.stream(). //
+					filter(jiraReleaseVersion -> jiraReleaseVersion.getName().equals(jiraVersion.toString())). //
+					forEach(jiraReleaseVersion -> versionsForModuleIteration.add(jiraReleaseVersion));
```

Comment:
Do we have to add them all in the first place if we going to select the first one eventually anyway?